src/pages/blog/neon.md

```diff
@@ -0,0 +1,495 @@
+```
+➜  note-taking git:(main) rails g model User database_information:json
+      invoke  active_record
+      create    db/migrate/20241012124645_create_users.rb
+      create    app/models/user.rb
+      invoke    test_unit
+      create      test/models/user_test.rb
+      create      test/fixtures/users.yml
+➜  note-taking git:(main) rails g devise User
+      invoke  active_record
+      create    db/migrate/20241012124742_add_devise_to_users.rb
+      insert    app/models/user.rb
+       route  devise_for :users
+```
+
+Next, we will run `rails db:migrate` to store our changes and add `database_information` to `config/initializers/filter_parameter_logging.rb`:
+```rb
+# Be sure to restart your server when you modify this file.
+
+# Configure parameters to be partially matched (e.g. passw matches password) and filtered from the log file.
+# Use this to limit dissemination of sensitive information.
+# See the ActiveSupport::ParameterFilter documentation for supported notations and behaviors.
+Rails.application.config.filter_parameters += [
+  ...,
+  :database_information,
+]
+```
+
+So now we have our users, but what about our notes? There are several advantages here to a multi-tenant architecture:
+
+- **Security by Design:** There is significantly less risk of harm due to accidentally destroying/fetching other customers data. This is because instead of worrying about multiple columns when you do things to a users notes, you just have to think about one. Additionally, you don't have to worry about the risk of someone finding another users notes by decrementing their own note ID since the users notes are pulled from the signed in record.
+- **Data Compliance:** If in the future we wanted to deploy in a region that the user specifies for compliance reasons (for example, an American user might want to deploy in America if you have a point of presence there due to latency reasons, but a European user might want to deploy in Europe due to Neon's GDPR compliance), this is a drop down box away with this architecture. Additionally, data compliance is significantly easier here. Want to add GDPR exports? Just dump out the entire database. Want to delete a user without worrying about errors due to foreign keys within your large database? Just delete the database (and we will tie this to the User record in this tutorial).
+- **Competitive Advantage:** With the multi-database model, you don't have to worry about leaking information that whilst it may not be sensitive to your users, is quite revealing about your company. For example, in this case, you might be able to figure out the companies user retention by taking a publicly sent customer count and then comparing it to the note ID since it is auto-incrementing with ActiveRecord.
+
+The main tradeoff with this model is that migrations become more complex, although this can be managed with careful branching and scripting.
```

Comment:
```suggestion
The main tradeoff with this model is that migrations become more complex, although we'll show how this can be managed with careful branching and scripting.
```